src/features/textEditor/components/tipTapNodes/citation/citationNode.ts

```diff
@@ -0,0 +1,193 @@
+      return null;
+    }
+
+    // Otherwise, if there was a replace step that added something in the citation node, we insert '; ' in the document
+    if (
+      transactions.find((transaction) =>
+        transaction.steps.find(
+          (step) => step instanceof ReplaceStep && step.from === step.to && step.from === oldPluginState,
+        ),
+      )
+    ) {
+      tr.insertText('; ', oldPluginState);
+      return tr;
+    }
+
+    return null;
+  },
+
+  props: {
+    decorations(state) {
+      const decorationPos = this.getState(state) as CitationPluginState;
+      if (!decorationPos) {
+        return null;
+      }
+      return DecorationSet.create(state.doc, [
+        Decoration.widget(
+          decorationPos,
+          () => {
+            const parentNode = document.createElement('span');
+
+            parentNode.innerHTML = '; ';
+            parentNode.style.color = 'hsl(var(--color-muted))';
+
+            return parentNode;
+          },
+          {
+            side: -1,
+          },
+        ),
+      ]);
+    },
+  },
+});
+
+/**
+ * Plugin that prevents the cursor from being directly after the opening and closing brackets of citation nodes
+ */
+const moveCursorPluginKey = new PluginKey('moveCursor');
+const moveCursorPlugin = new Plugin({
+  key: moveCursorPluginKey,
+  appendTransaction: (_transactions, oldState, newState) => {
+    if (!newState.selection.empty) {
+      return null;
+    }
+
+    const { doc, schema, selection, tr } = newState;
+    const { $from } = selection;
+
+    const isAfterOpeningBracket = $from.parent.type.name === citationNode.name && $from.nodeBefore === null;
+    const isAfterClosingBracket = $from.nodeBefore?.type.name === citationNode.name;
+
+    if (!isAfterOpeningBracket && !isAfterClosingBracket) {
+      return null;
+    }
+
+    const wasAfterBeforeOpeningBracket =
+      oldState.selection.empty && oldState.selection.$from.nodeAfter?.type.name === citationNode.name;
+
+    // If the the cursor was before the opening bracket and is now after the bracket,
+    // it means the user is trying to enter the node so we move the cursor inside
+    if (wasAfterBeforeOpeningBracket && isAfterOpeningBracket) {
+      tr.setSelection(TextSelection.create(tr.doc, $from.pos + 1));
+    } else {
+      // If there no text node in front of the citation node, create an "empty" one
+      // Note: prosemirror does not support empty text node, so we have to create a node with a space
+      if (isAfterOpeningBracket && !doc.resolve($from.before()).nodeBefore) {
+        tr.insert($from.before(), schema.text(' '));
+      }
+
+      tr.setSelection(TextSelection.create(tr.doc, $from.pos - 1));
+    }
+
+    return tr;
+  },
+});
+
+export const citationNode = Node.create({
+  name: 'citation',
+
+  content: '(text | reference)*',
+  defining: true,
+  group: 'inline',
+  inline: true,
+  selectable: true,
+
+  parseHTML: () => [
+    {
+      tag: 'citation',
```

Comment:
That makes sense! I'm changing the reference node to render as a `cite` instead of `span`. Should I reset the style? Currently `cite` have `font-style: italic`